x/incentives/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,248 @@
+package keeper_test
+
+import (
+	"time"
+
+	"github.com/c-osmosis/osmosis/x/incentives/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (suite *KeeperTestSuite) TestGRPCPotByID() {
+	suite.SetupTest()
+
+	// create a pot
+	addr1 := sdk.AccAddress([]byte("addr1---------------"))
+	startTime := time.Now()
+	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+	potID := suite.CreatePot(addr1, coins, types.DistrCondition{}, startTime, 2)
+
+	// not available pot
+	res, err := suite.app.IncentivesKeeper.PotByID(sdk.WrapSDKContext(suite.ctx), &types.PotByIDRequest{Id: 1000})
+	suite.Require().Error(err)
+	suite.Require().Equal(res, (*types.PotByIDResponse)(nil))
+
+	// final check
+	res, err = suite.app.IncentivesKeeper.PotByID(sdk.WrapSDKContext(suite.ctx), &types.PotByIDRequest{Id: potID})
+	suite.Require().NoError(err)
+	suite.Require().NotEqual(res.Pot, nil)
+	suite.Require().Equal(res.Pot.Id, potID)
+}
+
+func (suite *KeeperTestSuite) TestGRPCPots() {
+	suite.SetupTest()
+
+	// initial check
+	res, err := suite.app.IncentivesKeeper.Pots(sdk.WrapSDKContext(suite.ctx), &types.PotsRequest{})
+	suite.Require().NoError(err)
+	suite.Require().Len(res.Data, 0)
+
+	// create a pot
+	addr1 := sdk.AccAddress([]byte("addr1---------------"))
+	startTime := time.Now()
+	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+	suite.CreatePot(addr1, coins, types.DistrCondition{}, startTime, 2)
+
+	// final check
+	res, err = suite.app.IncentivesKeeper.Pots(sdk.WrapSDKContext(suite.ctx), &types.PotsRequest{})
+	suite.Require().NoError(err)
+	suite.Require().Len(res.Data, 1)
+}
+
+func (suite *KeeperTestSuite) TestGRPCActivePots() {
+	suite.SetupTest()
+
+	// initial check
+	res, err := suite.app.IncentivesKeeper.ActivePots(sdk.WrapSDKContext(suite.ctx), &types.ActivePotsRequest{})
+	suite.Require().NoError(err)
+	suite.Require().Len(res.Data, 0)
+
+	// create a pot
+	addr1 := sdk.AccAddress([]byte("addr1---------------"))
+	startTime := time.Now()
+	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+	suite.CreatePot(addr1, coins, types.DistrCondition{}, startTime, 2)
+
+	// final check
+	res, err = suite.app.IncentivesKeeper.ActivePots(sdk.WrapSDKContext(suite.ctx), &types.ActivePotsRequest{})
+	suite.Require().NoError(err)
+	suite.Require().Len(res.Data, 1)
```

Comment:
Ditto for checking that the returned pot is correct

Reply:
done